### ISSUE TITLE
Mobile navigation dialog makes mobile navigation content inert

### ISSUE THREAD

**Issue by jkjustjoshing** [uswds/uswds#4787]
### Describe the bug
In #3527, an issue was raised where the mobile navigation dialog was correctly trapping focus, but when using VoiceOver the focus could escape the mobile navigation dialog. This was fixed by adding an `aria-hidden` attribute to everything except the `.usa-header` element (but *only* if that header element is a direct descendant of `<body>`). 

However, this breaks in 2 scenarios, both of which are being demonstrated on https://designsystem.digital.gov. 
1. On https://designsystem.digital.gov itself, the mobile navigation contents, `.usa-nav`, is not a child of the `.usa-header` element, so `.usa-nav` is incorrectly getting an `aria-hidden` attribute. The result is that the focus correctly moves through the mobile navigation dialog when using the TAB key, but it does not when using VoiceOver to view the items in the navigation (e.g. VoiceOver Web Item rotor).
2. In the "Basic header" example on https://designsystem.digital.gov/components/header/ - [direct URL to example](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.app.cloud.gov/preview/uswds/uswds/develop/iframe.html?id=components-header--default&args=&viewMode=story) - the `.usa-header` element is _not_ a direct descendant of `<body>`, so its parent incorrectly has an `aria-hidden` attribute when the mobile navigation dialog is open.

### Example 1
1. Navigate to https://designsystem.digital.gov/
2. Shrink the page, or zoom in to the point where the mobile navigation styling is activated
3. Tab to and activate the menu button
4. In the developer tools element inspector, note that the `<header class="usa-header">` does not have an `aria-hidden` attribute, but note that the `<nav class="usa-nav site-nav sidenav-mobile">` _does_ have an `aria-hidden` attribute, despite being the content that the user is trying to interact with.

### Example 2
1. Navigate to [the Basic Header example](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.app.cloud.gov/preview/uswds/uswds/develop/iframe.html?id=components-header--default&args=&viewMode=story) 
2. Shrink the page, or zoom in to the point where the mobile navigation styling is activated
3. Tab to and activate the menu button
4. In the developer tools element inspector, note that the `<header class="usa-header">` element is a child of `<div id="root">`, and that element has an `aria-hidden` attribute.

### Expected behavior

All of the page content has `aria-hidden` applied to it, EXCEPT for the mobile navigation dialog's content.

This can be tested using VoiceOver, by
1. Using ctrl+option+➡️ key combination should cycle through all items in the mobile navigation dialog, but nothing else on the page.
2. Using ctrl+option+U, the VoiceOver Rotor, to view all the mobile navigation dialog's links, but no other links on the page.

This should be the case for a variety of `.usa-header` and `.usa-nav` document structures, since there are likely many different structures in the wild at this point.

**Comment by jkjustjoshing** [comment]
I initially considered logging this against the https://github.com/uswds/uswds-site repo, because arguably this could be fixed by updating the header structure in that site's markup. However because the Basic Header example is also broken, it seems like a more universal fix at the library level might be warranted.
